lib/presentation/widget/myspace/my_space_widget.dart

```diff
@@ -627,4 +630,56 @@ class _MySpaceWidgetState extends State<MySpaceWidget> {
         .onActionClick((data) => mySpaceViewModel.previewDocument(context, document))
         .build();
   }
+
+  String _getIconSort(OrderType orderType) {
+    return orderType == OrderType.ascending ? imagePath.icSortUpCurrent : imagePath.icSortDownCurrent;
+  }
+
+  Widget _buildMenuSorter() {
+    return StoreConnector<AppState, AppState>(
+      converter: (Store<AppState> store) => store.state,
+      builder: (context, appState) {
+        return (!appState.uiState.isInSearchState() && appState.mySpaceState.documentList.isNotEmpty) ? ListTile(
+          leading: Transform(
+            transform: Matrix4.translationValues(5, 5, 0.0),
+            child: SvgPicture.asset(
+              _getIconSort(appState.mySpaceState.sorter.orderType),
+              width: 15,
+              height: 15,
+              fit: BoxFit.fill,
+              color: AppColor.primaryColor,
+            ),
+          ),
+          title: Transform(
+              transform: Matrix4.translationValues(-25, 0.0, 0.0),
+              child: Text(appState.mySpaceState.sorter.orderBy.getName(context),
+                maxLines: 1,
+                style: TextStyle(fontSize: 14, color: AppColor.documentNameItemTextColor),
+              )
+          ),
+          tileColor: AppColor.topBarBackgroundColor,
+          onTap: () => {mySpaceViewModel.openPopupMenuSorter(context, _listSorterActions(context, appState.mySpaceState.sorter))},
+        ) : SizedBox.shrink();
+      },
+    );
+  }
+
+  List<Widget> _listSorterActions(BuildContext context, Sorter currentSorter) {
```

Comment:
I think you can use a map there, think about it

Reply:
Cool. I will try it.